doc/general_design.md

```diff
@@ -73,7 +73,7 @@ Depends on the demo I wrote, cipher suites should be `TLS13_AES_256_GCM_SHA384`.
 
 #### cgroups ####
 
-After jobs run, their pid should add to specific cgroup if its all parent processs have added to cgroup.
+After jobs run, their pid should add to specific cgroup before it can folk..
```

Comment:
How will you accomplish this?

Reply:
Wait, look like it is impossible. I am trying to answer your question if the job fork before adding to cgroup. I haven't paly around cgroup yet. Let me do some study about cgroup. Then I will change this part of the doc.

Reply:
Hi Zac, I play a bit of cgroup and it looks like if I move parent's pid to specific cgroup, their children don't follow it. I think I can get all children and move them into cgroup, then move children's children, and so on...

Then I notice if the parent processing has a `folk bomb`, this recursive moving function will run out resources sooner or later. So, should I just assume the job commands the client sent to the server don't have `folk bomb` inside?

Reply:
I don't think that's acceptable as a solution, it's not some weird thing that only happens if a program is malicious: if you're using this to launch some management shell script and the script launches some background thing immediately, that thing will end up outside of the cgroup.

You could check out how other cgroup-aware process management supervisors do things - most notably systemd.

Reply:
Thanks! Let me check.